### PR TITLE
fix: Update browser navigation on directory links

### PR DESCRIPTION
**Pull request by acezard**
Related to: https://trello.com/c/BdnfOXYA/1510-%F0%9F%93%81-drive-le-lien-sur-le-chemin-du-dossier-ne-fonctionne-pas-bien

Issues:
- note directory opens a note instead of its dir
- other files open the dir, but the preview tool when going back

Bugs:

- conflict with FileName onClick() and Link in Dir Name (link inside a link => double routing)

Resolution:
- add a guard on file opening method

### CODE REVIEW

File: src/drive/web/modules/filelist/FileOpener.jsx [hunk 1]
```diff
@@ -80,6 +80,10 @@ const FileOpener = ({
             ev.srcEvent.stopImmediatePropagation()
             toggle(ev.srcEvent)
           } else {
+            // We need to filter here, is the event target the FileOpener or a link INSIDE the FileOpener?
+            // Assuming the FileOpener itself will never be inside an anchor, we can use the following guard
+            if (ev.srcEvent.target.closest('a')) return
```
Comment (Crash--) [inline src/drive/web/modules/filelist/FileOpener.jsx:85]: Should we put this logic in enableTouchEvents() method?

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:85]: hum yeah I guess so, we can put it in the first guard (L32)

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:85]: actually line 45 was trying to do that it seems, I guess it's not working anymore for some reason

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx]: updated

File: package.json [hunk 2]
```diff
@@ -159,7 +159,7 @@
     "react-measure": "2.5.0",
     "react-redux": "7.2.0",
     "react-remove-scroll": "^2.4.1",
-    "react-router": "3.2.4",
+    "react-router": "3.2.6",
```
Comment (Crash--) [inline package.json:162]: should be done in another commit.

File: src/drive/web/modules/filelist/FileOpener.jsx [hunk 3]
```diff
@@ -42,14 +42,9 @@ const enableTouchEvents = ev => {
...
-  const parentLink = getParentLink(ev.target)
-  if (
-    parentLink &&
-    parentLink.className.indexOf(styles['fil-file-path']) >= 0
-  ) {
-    return false
-  }
+  // We need to filter here, is the event target the FileOpener or a link INSIDE the FileOpener?
+  // Assuming the FileOpener itself will never be inside an anchor, we can use the following guard
+  if (ev.srcEvent.target.closest('a')) return false
```
Comment (Crash--) [inline src/drive/web/modules/filelist/FileOpener.jsx:47]: I know that there is no test for this method, but I think this is something we should add. We have too many regressions on this. What do you think?

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: I don't know it's hard to test, would basically needs to render a whole DOM tree up to window. I'm basing myself on the assumption that it's impossible to find an anchor parent in the event unless it was launched by a child node, ie router link in this case. I tried to not be specific but I could also use a class on the folder link. But then the bug could happen elsewhere if some link is added

File: package.json (hunk 2 above)
Comment (acezard) [reply package.json:162]: updated

File: src/drive/web/modules/filelist/FileOpener.jsx (hunk 3 above)
Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: I guess FileOpener having anchor as a parent should be a bug anyway, it would not be valid HTML

Comment (Crash--) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: > I don't know it's hard to test, would basically need to render a whole DOM tree up to window. I'm basing myself on the assumption that it's impossible to find an anchor parent in the event unless it was launched by a child node, ie router link in this case. I tried to not be specific but I could also use a class on the folder link. But then the bug could happen elsewhere if some link is added

We already have a DOM test for /recent view. We already do some click on a few actions. I don't think it would be difficult to test with that, no?

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: I guess it's possible sure, maybe verifying react-router is only invoked once. Or creating stubs and see where react-router sends me on different clicks. I'm worrying about delivery time though, my tests are failing in local and I don't know why. I need to fix this and then write the tests. I'd propose to do this in another PR if at all possible

Comment (Crash--) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: >  I need to fix this and then write the tests. I'd propose to do this in another PR if at all possible

Since the test file already exist, already display a File within the view with a path displayed I think, we should be able to click on the path and check the new url. I think this is mandatory.

Comment (Crash--) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: You can also only test this method by creating an event with different kind of targets / html .

Comment (acezard) [reply src/drive/web/modules/filelist/FileOpener.jsx:47]: I added two expects (I would prefer to have multiple tests but this test is already using a pattern with a lot of assert in a single test). 

I also add a small prettier file to match the eslint config for IDE purposes, not just mine but everyone having Prettier plugin + VSCODE for instance. Is that ok?

**Review by Crash-- — APPROVED** [review]
Thanks for the effort
